### PR TITLE
fixed bug where the same task got killed multiple times

### PR DESCRIPTION
**Pull request by chaudum**

### CODE REVIEW

File: src/main/java/io/crate/frameworks/mesos/CrateExecutor.java [hunk 1]
```diff
@@ -180,6 +172,10 @@ public void killTask(ExecutorDriver driver, TaskID taskId) {
     }
 
     private void gracefulShutdownCrate(ExecutorDriver driver) {
+        driver.sendStatusUpdate(TaskStatus.newBuilder()
```
Comment (kovrus) [inline src/main/java/io/crate/frameworks/mesos/CrateExecutor.java]: We can try to move `sendStatusUpdate` and the `TaskStatus` builder into its own method. The same code block repeats few times, but the only difference is the task state.

File: src/main/java/io/crate/frameworks/mesos/CrateExecutor.java [hunk 2]
```diff
@@ -155,21 +155,13 @@ public void launchTask(ExecutorDriver driver, TaskInfo taskInfo) {
...
         LOGGER.info("Killing task : " + taskId.getValue());
         healthCheck.cancel(true);
-        if(forceShutdown){
+        int pid = task.pid();
+        if (forceShutdown){
             forceShutdownCrate(driver);
-            return;
-        }
-        if (task.process != null) {
+        } else if (pid >= 0 && task.process != null) {
```
Comment (kovrus) [inline src/main/java/io/crate/frameworks/mesos/CrateExecutor.java]: 0 and 1 PIDs are reserved?

**Comment by kovrus** [comment]
otherwise ack

File: src/main/java/io/crate/frameworks/mesos/CrateExecutor.java (hunk 2 above)
Comment (chaudum) [inline src/main/java/io/crate/frameworks/mesos/CrateExecutor.java]: `task.pid()` returns `-1` if pid not found
everything else it returns is OK

**Comment by kovrus** [comment]
ack

**Pull request merged**